### PR TITLE
Moves the jenkins scripts out of root into subdirectory

### LINKED ISSUE #48: Move jenkins.*.* scripts into a jenkins directory off the project root.
This will make the contents of the root of our project more legible, and effectively bring the `README` into view sooner on the main page of the project.

It will require modifications to most of those scripts and to the `netci.groovy` script itself.

The build checks will break on the PR for this change because the Jenkins task definitions will be out of sync with the actual locations of the files, so we will need to accept the change despite the breaks and quickly react to breaks following that change.

### PR DESCRIPTION
**Pull request by ianwjhalliday**
Fix #48 

- Move the jenkins scripts to a new directory named `jenkins`.
- Remove the `jenkins.` prefix from their file names since it is now redundant.
- Update the scripts to reflect the moved and renamed files

Note `netci.groovy` must remain in the root for CI to work.

There is no change to the `test/jenkins.*` files.  We could consider consolidating them too, but for now my motivation is to clean out the root directory.

### CODE REVIEW

**Comment by ianwjhalliday** [comment]
@dilijev cc @mmitche

**Comment by ianwjhalliday** [comment]
I'll squash the commits after review.

**Comment by ianwjhalliday** [comment]
Re

> For the linux scripts as defined, you'll want to make sure that the scripts are invoked relative to the working directory and not the script directory. I don't know if linux has an equivalent of %~dp0 for that purpose.

I haven't verified this yet.  It does appear that *nix shells do not have a %~dp0 equivalent.

File: jenkins/buildall.cmd [hunk 1]
```diff
@@ -7,6 +7,6 @@
 setlocal
 
 set JENKINS_BUILD=True
-call test\jenkins.testall.cmd %*
```
Comment (dilijev) [inline jenkins/buildall.cmd]: No problem here, I just think it's goofy that Git/GitHub detected these two files as renames of each other. I guess the diff is the same size regardless of whether the script name changed.

**Comment by dilijev** [comment]
Re squash: you should actually make two commits to help git. In the first, do all the renaming (and in the same commit is also a good place to preserve the execute bit). Then, actual code changes in the second.

File: netci.groovy [hunk 2]
```diff
@@ -163,5 +163,5 @@ CreateBuildTasks('Windows_NT', 'daily_disablejit', ' "/p:BuildJIT=false"', ' -di
 // CODE STYLE TASKS
 // ----------------
 
-CreateStyleCheckTasks('./jenkins.check_eol.sh', 'ubuntu_check_eol', 'EOL Check')
-CreateStyleCheckTasks('./jenkins.check_copyright.sh', 'ubuntu_check_copyright', 'Copyright Check')
+CreateStyleCheckTasks('./jenkins/check_eol.sh', 'ubuntu_check_eol', 'EOL Check')
+CreateStyleCheckTasks('./jenkins/check_copyright.sh', 'ubuntu_check_copyright', 'Copyright Check')
```
Comment (dilijev) [inline netci.groovy]: Good. For this reason, we expect the checks to fail. We should discuss how to bypass the checks and bring this in. Now might be a good time to add a checkbox to bypass the checks.

**Comment by mmitche** [comment]
LGTM

**Comment by dilijev** [comment]
@dotnet-bot test ci please

**Comment by dilijev** [comment]
@ianwjhalliday cmd scripts are subject to the copyright check as well :P

**Comment by dilijev** [comment]
@dotnet-bot test ci please

**Comment by dilijev** [comment]
@ianwjhalliday LGTM. I'll sign off when you finish the rebase.

**Comment by dilijev** [comment]
Mostly LGTM after rebase. I confirmed the execute bits are set on all the sh scripts. CI was successful.

Only problem is I just merged #414 and now you've got a conflict. Sorry about that.

**Comment by dilijev** [comment]
@dotnet-bot test ci please

**Comment by dilijev** [comment]
@ianwjhalliday signed off. Good to merge when checks pass.

**Pull request merged**